### PR TITLE
Added the `require-event-dispatcher-types` rule

### LINKED ISSUE #346: New rule: `require-event-dispatcher-types`
### Motivation

Hi,
when using svelte with TypeScript, it is possible to add types to events by typing the `createEventDispatcher` function so:

```svelte
import { createEventDispatcher } from "svelte";

const dispatch = createEventDispatcher<{one: never, two: number}>();

dispatch("one"); // OK
dispatch("two", 3); // OK
dispatch("one", "Hello world"); // Fails
dispatch("two"); // Fails
dispatch("two", "Hello world"); // Fails
```

I propose a rule that would require all calls to `createEventDispatcher` to include the templated type.

### Description

The rule would trigger on any call of `createEventDispatcher` witchou the templated type

### Examples

```svelte
import { createEventDispatcher } from "svelte";

<!-- ✓ GOOD -->
const dispatch = createEventDispatcher<{one: never, two: number}>();

const dispatch = createEventDispatcher<any>();

const dispatch = createEventDispatcher<{}>();

<!-- ✗ BAD -->
const dispatch = createEventDispatcher();
```

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by marekdedic**
Closes #346

### CODE REVIEW

File: src/rules/require-event-dispatcher-types.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    schema: [],
+    messages: {
+      missingTypeParameter: `Type parameters missing for the createEventDispatcher function call.`,
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    return {
+      Program() {
```
Comment (ota-meshi) [inline src/rules/require-event-dispatcher-types.ts:19]: I think we need to check if `lang="ts"` is specified before we check it.

```suggestion
    let isTs = false
    return {
      SvelteScriptElement(node) {
        const lang = getLangValue(node)?.toLowerCase()
        if (lang === "ts" || lang === "typescript") {
          isTs = true
        }
      },
      "Program:exit"() {
        if (isTs) return
```

File: tests/fixtures/rules/require-event-dispatcher-types/invalid/no-types01-input.svelte [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<script lang="ts">
```
Comment (ota-meshi) [inline tests/fixtures/rules/require-event-dispatcher-types/invalid/no-types01-input.svelte:1]: Could you add another test case without `lang="ts"`?

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR! I made some comments.

**Comment by marekdedic** [comment]
Thanks for the review :) It's a good catch that the rule should probably ignore non-ts code outright.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**